pilot/pkg/proxy/envoy/v2/debounced_channel.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"sync"
+	"time"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+// DebouncedEvent is used as entry type, which is enqueued in DebouncedChannel
+type DebouncedEvent = interface{}
+
+// DebouncedChannel is a channel which abstracts the debounce algorithm
+type DebouncedChannel struct {
+	// C channel, which can be used in selects. It delivers debounced events.
+	// The function emitted must be called. Otherwise processing is stopped.
+	C <-chan func(func(DebouncedEvent))
```

Comment:
This double function callback style is very complex. A quick grep indicates we actually never do this anywhere in istio today. I think it would be clearer if instead we just have a `chan Foo` and a 
```
struct Foo {
   event DebounceEvent
   done func()
}
```

I find this much more explicit. What do you think?

I'll leave the naming up to you :slightly_smiling_face:

Reply:
OK. We will call it `Notification`